### ISSUE TITLE
microsoft/iis:nanoserver crashes when running ServiceMonitor

### ISSUE THREAD

**Issue by FFLSH** [Microsoft/iis-docker#47]
To reproduce:

Create and build a Dockerfile as in the example on the microsoft/iis docker hub page

```
# escape = `

FROM microsoft/iis:nanoserver

RUN mkdir C:\app

RUN powershell -NoProfile -Command `
    Import-module IISAdministration; `
    New-IISSite -Name "Site" -PhysicalPath C:\app -BindingInformation "*:8000:"

EXPOSE 8000
```

At this point you could also copy site contents into `C:\app` - it doesn't make a difference

When running the image via `docker run` the following is presented:

```
ERROR ( message:Cannot find requested collection element. )
Applied configuration changes to section "system.applicationHost/applicationPools" for "MACHINE/WEBROOT/APPHOST" at configuration commit path "MACHINE/WEBROOT/APPHOST"

 Service 'w3svc' has been stopped

APPCMD failed with error code 4312

ERROR: Failed to start or quesry status of service 'w3svc' error [80004005]

 Service 'w3svc' has been stopped
```

The container then exits as `ServiceMonitor` has crashed.

**Comment by mcy94w** [comment]
I just merge the new version of servicemonitor to master, could you try the new version?

**Comment by FFLSH** [comment]
@mcy94w 
I have just tried that, unfortunately I still get the same error message.

**Comment by FFLSH** [comment]
Any ideas? This is blocking our deployments of Angular apps.

**Comment by shirhatti** [comment]
Do you have a repro you can share? Either push your image to a registry or share a tarball containing the container image with us?

**Comment by FFLSH** [comment]
@shirhatti If you look at the OP I included reproduction steps.

**Comment by FFLSH** [comment]
@shirhatti I've pushed an image with the issue up to [here](https://hub.docker.com/r/fletech/47/) so you can run it by doing `docker run -d -p 8000:8000 fletech/47:1` and observe the issue.

**Comment by FFLSH** [comment]
Looks like [this](https://github.com/Microsoft/IIS.ServiceMonitor/issues/4) is related or the same. Does `microsoft/iis` use that repository?

**Comment by shirhatti** [comment]
Yes. `microsoft/iis` uses ServiceMonitor. It appears to be the same issue. We're looking into it now

**Comment by putz612** [comment]
Any issues on this?  I am experiencing this as well using the latest version.

**Comment by shirhatti** [comment]
@putz612 This issue was fixed in the September release. See linked PR.
Are you still experiencing this issue? If so, what image are you running?

**Comment by putz612** [comment]
Just created this yesterday pulling nanoserver:latest so that would be nanoserver-10.0.14393.1770.  I am just creating a POC using the example in hub.docker so my Dockerfile is real simple.

```
FROM microsoft/iis:nanoserver

RUN mkdir C:\site

RUN powershell -NoProfile -Command \
    Import-module IISAdministration; \
    New-IISSite -Name "Site" -PhysicalPath C:\site -BindingInformation "*:8000:"
	
EXPOSE 8000

ADD index.html /site
```

When I start it I get the same issue as above.
```
ERROR ( message:Cannot find requested collection element. )
Applied configuration changes to section "system.applicationHost/applicationPools" for "MACHINE/WEBROOT/APPHOST" at configuration commit path "MACHINE/WEBROOT/APPHOST"

 Service 'w3svc' has been stopped

APPCMD failed with error code 4312

ERROR: Failed to start or quesry status of service 'w3svc' error [80004005]

 Service 'w3svc' has been stopped
Applied configuration changes to section "system.applicationHost/applicationPools" for "MACHINE/WEBROOT/APPHOST" at configuration commit path "MACHINE/WEBROOT/APPHOST"
Applied configuration changes to section "system.applicationHost/applicationPools" for "MACHINE/WEBROOT/APPHOST" at configuration commit path "MACHINE/WEBROOT/APPHOST"

 Service 'w3svc' has been stopped

ERROR: Failed to start or quesry status of service 'w3svc' error [80004005]

 Service 'w3svc' has been stopped
```

I am on windows 2016 server with the latest patches, docker info below.

```
Containers: 1
 Running: 0
 Paused: 0
 Stopped: 1
Images: 9
Server Version: 17.06.1-ee-2
Storage Driver: windowsfilter
 Windows:
Logging Driver: json-file
Plugins:
 Volume: local
 Network: l2bridge l2tunnel nat null overlay transparent
 Log: awslogs etwlogs fluentd json-file logentries splunk syslog
Swarm: inactive
Default Isolation: process
Kernel Version: 10.0 14393 (14393.1770.amd64fre.rs1_release.170917-1700)
Operating System: Windows Server 2016 Standard
OSType: windows
Architecture: x86_64
CPUs: 2
Total Memory: 8GiB
```

**Comment by bariscaglar** [comment]
We are able to repro this on our end and will look into it. Let me point out that we have not released a IIS Nano container image for 1709 and will likely not release IIS on Nano in the foreseeable future. The application model is shifting towards using Kestrel for .NET Core applications on Nano and not use IIS. If you'd like to use IIS, ServerCore is the current recommended SKU.

**Comment by sirdank** [comment]
I'm getting this on plain microsoft/iis non-nano. Trying to find a minimal repro.

**Comment by renatotkr** [comment]
same here with both iis and iis nanoserver

**Comment by bariscaglar** [comment]
@mcy94w any updates so far?

**Comment by mcy94w** [comment]
This should be a nano only issue.
We fixed in https://github.com/Microsoft/iis-docker/pull/73.

**Comment by bariscaglar** [comment]
That means our next Nano image (February) should work?

**Comment by minoleg** [comment]
I am getting the same issue running container "microsoft/iis:windowsservercore-ltsc2016" on host "Microsoft Windows Server Version 1607 (OS Build 14393.1715)".

Is there a solution or workaround by any chance?

Thanks.

**Comment by mcy94w** [comment]
@minoleg We updated the nano image on Feb13 which should fix this issue. Could you try again with the new images?

**Comment by michal-clapinski** [comment]
I'm also sometimes (around 30% of cases) getting this error, even on the newest images (`microsoft/iis:latest` from yesterday). I've downgraded to `microsoft/iis:windowsservercore-10.0.14393.1358` and it works fine.

**Comment by LukeRH** [comment]
I also had this error and like michal-clapinski downgrading to microsoft/iis:windowsservercore-10.0.14393.1358 fixed it.

**Comment by bariscaglar** [comment]
@LukeRH @michal-clapinski  what is "This error" in this case? Can you give more details. We are trying to pinpoint the issue and believe that the april image might fix it. I want to make sure that we are all talking about the same problem.

**Comment by LukeRH** [comment]
@bariscaglar The error message we had was "Failed to stop or query status of service 'w3svc' error [800705b4] " Unfortunately i no longer have the log files. I have reverted to use latest iis docker image in our docker image hopefully then i can reproduce this issue and provide you with further information.

**Comment by mcy94w** [comment]
@LukeRH :

The fix has been made to ServiceMonitor.exe to resolve the issue "Failed to stop or query status of service 'w3svc'". The new images has been published Yesterday. Please update the image and let us know if you still see this error.

**Comment by mcy94w** [comment]
Hi @michal-clapinski :

We have release the new image which contains the fix of this issue. https://github.com/Microsoft/IIS.ServiceMonitor/pull/29

**Comment by shirhatti** [comment]
@robertoandrade What version of `microsoft/iis` are you using? Can you share the image digest `docker images --no-trunc`

**Comment by robertoandrade** [comment]
```
REPOSITORY                    TAG                                 IMAGE ID                                                                  CREATED             SIZE
microsoft/iis                 latest                              sha256:b8f924611ebbfb7424bbc262f67aff4d0cf1aca74027b4b5e1ecbd329f5425de   2 weeks ago         10.8GB
```

**Comment by JoseFMP** [comment]
For me this issue is still happening with the latest image by today.

**Comment by bariscaglar** [comment]
@JoseFMP we no longer support microsoft/iis:nanoserver images. If your problem is elsewhere, please file another issue as this one is already closed.
